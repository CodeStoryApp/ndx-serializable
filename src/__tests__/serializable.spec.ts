import { createIndex, addDocumentToIndex } from "ndx";
import { toSerializable, fromSerializable } from "..";

const tokenizer = (s: string) => s.split(" ");
const filter = (s: string) => s;

test("should convert to serializable data structure", () => {
  const idx = createIndex<number>(2);
  const docs = [
    { id: 1, title: "a b c", text: "hello world" },
    { id: 2, title: "c d e", text: "lorem ipsum" },
  ];
  docs.forEach((doc) => {
    addDocumentToIndex(idx, [(d) => d.title, (d) => d.text], tokenizer, filter, doc.id, doc);
  });
  expect(toSerializable(idx)).toMatchSnapshot();
});

test("should convert from serializable data structure", () => {
  const idx = createIndex<number>(2);
  const docs = [
    { id: 1, title: "a b c", text: "hello world" },
    { id: 2, title: "c d e", text: "lorem ipsum" },
  ];
  docs.forEach((doc) => {
    addDocumentToIndex(idx, [(d) => d.title, (d) => d.text], tokenizer, filter, doc.id, doc);
  });
  expect(fromSerializable(toSerializable(idx))).toEqual(idx);
});
