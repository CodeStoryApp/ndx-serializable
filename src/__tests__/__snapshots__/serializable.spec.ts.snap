// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should convert to serializable data structure 1`] = `
Object {
  "docFieldLengths": Array [
    Array [
      3,
      2,
    ],
    Array [
      3,
      2,
    ],
  ],
  "docIds": Array [
    1,
    2,
  ],
  "fields": Array [
    Object {
      "avg": 3,
      "sum": 6,
    },
    Object {
      "avg": 2,
      "sum": 4,
    },
  ],
  "postings": Array [
    Array [
      Array [
        1,
        Array [
          1,
          0,
        ],
      ],
    ],
    Array [
      Array [
        1,
        Array [
          1,
          0,
        ],
      ],
    ],
    Array [
      Array [
        1,
        Array [
          1,
          0,
        ],
      ],
      Array [
        2,
        Array [
          1,
          0,
        ],
      ],
    ],
    Array [
      Array [
        1,
        Array [
          0,
          1,
        ],
      ],
    ],
    Array [
      Array [
        1,
        Array [
          0,
          1,
        ],
      ],
    ],
    Array [
      Array [
        2,
        Array [
          1,
          0,
        ],
      ],
    ],
    Array [
      Array [
        2,
        Array [
          1,
          0,
        ],
      ],
    ],
    Array [
      Array [
        2,
        Array [
          0,
          1,
        ],
      ],
    ],
    Array [
      Array [
        2,
        Array [
          0,
          1,
        ],
      ],
    ],
  ],
  "terms": Array [
    "a",
    "b",
    "c",
    "hello",
    "world",
    "d",
    "e",
    "lorem",
    "ipsum",
  ],
}
`;
